### PR TITLE
2d interpolation for dfsu

### PR DESCRIPTION
**Pull request by jsmariegaard**

### CODE REVIEW

File: mikeio/dfsu.py [hunk 1]
```diff
@@ -488,6 +499,119 @@ def _get_element_coords(self):
...
+        x = np.linspace(x0, x1, nx)
+        y1 = y0 + dy*(ny-1)
+        y = np.linspace(y0, y1, ny)
+        xv, yv = np.meshgrid(x, y)
+        xy = np.empty((nx*ny, 2))
+        xy[:,0] = xv.reshape(-1)
+        xy[:,1] = yv.reshape(-1)
+        return self.get_2d_interpolant(xy, n_nearest)
+
+    def interp2d(self, data, interpolant2d):
```
Comment (ecomodeller) [inline mikeio/dfsu.py:598]: interp2d could be extracted into a new class

Comment (jsmariegaard) [reply mikeio/dfsu.py:598]: good idea

File: mikeio/dfsu.py [hunk 2]
```diff
@@ -488,6 +499,119 @@ def _get_element_coords(self):
...
+                nx = int(np.ceil(xr/yr))*ny
+            dx = xr/(nx-1)
+            dy = yr/(ny-1)
+        else:
+            if dx is None:
+                nx, ny = shape
+                dx = xr/(nx-1)
+                dy = yr/(ny-1)
+            if shape is None:
+                dy = dx
```
Comment (ecomodeller) [inline mikeio/dfsu.py:557]: Allow dx to be either None, float or a two-item list or tuple to get different dx and dy

Comment (jsmariegaard) [reply mikeio/dfsu.py]: Done

File: mikeio/dfsu.py (hunk 1 above)
Comment (jsmariegaard) [reply mikeio/dfsu.py:623]: I have now extracted functionality into a new file - but could not really see the benefit of making it a class - maybe you can?

Comment (ecomodeller) [reply mikeio/dfsu.py]: No.

Comment (jsmariegaard) [reply mikeio/dfsu.py]: Okay - but I'll guess we will keep it in a separate file for now, right?

**Pull request merged**